.github/workflows/debug.yml

```diff
@@ -2,10 +2,17 @@
 name: Debug
 
 'on':
-  [push, pull_request, pull_request_review, release, workflow_dispatch, workflow_call]
+  push:
+    branches:
+      # 22.1 and 22.10
+      - '2[1-9].[1-9][0-9]'
+      - '2[1-9].[1-9]'
```

Comment:
I'd like to see all the branches here, as it used to be. Why is it better to place them in many places instead of a single one?

The disadvantage of decoupling it that's now it's a dedicated job, and then it will be hidden *somewhere*

Reply:
I don't want Debug wf to pop up in each PR or push, because it's needed on rare occasions. For PR and push to master it is still available inside RunConfig job.

Reply:
Now:
![image](https://github.com/ClickHouse/ClickHouse/assets/3025537/a4ca5a02-0aea-4adf-8cd8-cb52f6e08276)

after the changes:
![image](https://github.com/ClickHouse/ClickHouse/assets/3025537/49287169-69b3-4333-b169-de35dbba2d65)

Reply:
why we need it on pull_request_review? We don't have any workflow with this event. I propose to remoe it for pull_request_review as well